### ISSUE TITLE
problems with dns handling and priority, a follow up

### ISSUE THREAD

**Issue by xioren** [mullvad/mullvadvpn-app#3418]
## Issue report

Operating system: Debian Testing

App version: 2022.1

### Issue description

I previously opened an issue #3139 where the wireguard dns was getting tacked onto the end of the resolv.conf file and thus being ignored as there is a limit to the number of resolvers that are honored in that file. I run into this issue on my laptop but not my desktop and upon further inspection I noticed something interesting. On my pc, where I do not have this problem, Mullvad replaces the resolv.conf with a new file containing mullvads resolver and backs up the old file as resolv.conf.mullvadbackup. This file is not created on my laptop and instead the Mullvad resolver is appended to the end of the current list and subsequently ignored. The os on both systems is identical, the Mullvad app version is identical as well. I am wondering why the discrepancy. What would cause Mullvad not to create the backup file, permission issue?

**Comment by pinkisemils** [comment]
The issue here is that on your laptop you have [`resolvconf`](https://packages.debian.org/sid/resolvconf) package installed and our client is using it to configure DNS. You can force a particular way of managing DNS via [setting](https://github.com/mullvad/mullvadvpn-app#setting-environment-variable) an [environment variable](https://github.com/mullvad/mullvadvpn-app#environment-variables-controlling-the-execution) for our daemon, specifically the `TALPID_DNS_MODULE`. 

In my experience, unless a distribution has gone out of it's way to deliver artisanal BOFH behavior for `systemd-resolved`, it seems to be able to handle per-interface DNS configurations most reliably.

**Comment by xioren** [comment]
Yes your're right, my laptop does have resolvconf installed. It is using network manager. It's all default Debian, I haven't made any major changes.

Which option do you recommond for the environment variable?

static-file", "resolvconf", "systemd", "network-manager"

**Comment by xioren** [comment]
For what it's worth, doing some reading an I came to the conclusion that the way resolv.conf was set up was not the recommended way (despite it being that way by default in Debain). Instead of resolv.conf being a static file, it should be a sim-link from /run/systemd/resolve/stub-resolv.conf. Upon doing this (and removing openresolv, not sure how it got installed) everything is working as expected. No need for setting an environment variable.
